updates for 2.6

djdanielsson opened the pull request:
<!-- markdownlint-disable -->
# What does this PR do?
<!--- Brief explanation of the code or documentation change you've made -->
adding gateway_role_team_assignments and gateway_role_definitions and updating gateway_role_user_assignments

# How should this be tested?
<!--- Automated tests are preferred, but not always doable - especially for infrastructure. Include commands to run your new feature, and also post-run commands to validate that it worked. (please use code blocks to format code samples) -->
manually

# Is there a relevant Issue open for this?
<!--- Provide a link to any open issues that describe the problem you are solving. -->
no

# Other Relevant info, PRs, etc
<!--- Please provide link to other PRs that may be related (blocking, resolves, etc. etc.) -->
none

djdanielsson committed:
new file:   gateway_role_definitions/defaults/main.yml
	new file:   gateway_role_definitions/meta/argument_specs.yml
	new file:   gateway_role_definitions/meta/main.yml
	new file:   gateway_role_definitions/tasks/main.yml
	new file:   gateway_role_team_assignments/defaults/main.yml
	new file:   gateway_role_team_assignments/meta/argument_specs.yml
	new file:   gateway_role_team_assignments/meta/main.yml
	new file:   gateway_role_team_assignments/tasks/main.yml
	modified:   gateway_role_user_assignments/defaults/main.yml
	modified:   gateway_role_user_assignments/meta/argument_specs.yml

djdanielsson committed:
adding some generated readme just to have something

djdanielsson committed:
Merge branch 'aap26_testing' into aap26_testing

djdanielsson committed:
Merge branch 'devel' into aap26_testing

djdanielsson committed:
modified:   roles/gateway_role_definitions/README.md
	modified:   roles/gateway_role_team_assignments/README.md
	modified:   roles/gateway_role_user_assignments/README.md

Tompage1994 reviewed (changes requested):
Looks good. Needs a changelog though

djdanielsson committed:
new file:   changelogs/fragments/aap_26_additions.yml
	modified:   roles/gateway_role_definitions/README.md
	modified:   roles/gateway_role_definitions/meta/argument_specs.yml
	modified:   roles/gateway_role_team_assignments/README.md
	modified:   roles/gateway_role_team_assignments/meta/argument_specs.yml
	modified:   roles/gateway_role_user_assignments/README.md

djdanielsson committed:
modified:   changelogs/fragments/aap_26_additions.yml

djdanielsson committed:
modified:   roles/gateway_role_definitions/README.md
	modified:   roles/gateway_role_team_assignments/README.md
	modified:   roles/gateway_role_user_assignments/README.md

djdanielsson commented on roles/gateway_role_definitions/meta/argument_specs.yml:
```diff
@@ -0,0 +1,100 @@
+  main:
+    short_description: An Ansible Role to create role_definitions on Ansible gateway for AAP 2.6+.
+    options:
+      aap_role_definitions:
```
do these arg_specs look good?

Tompage1994 replied:
Doesn't seem to match the variable you're actually using (`gateway_role_definitions`)

djdanielsson committed:
modified:   roles/gateway_role_definitions/README.md
	modified:   roles/gateway_role_team_assignments/README.md
	modified:   roles/gateway_role_user_assignments/README.md

Tompage1994 commented on roles/gateway_role_user_assignments/README.md:
```diff
@@ -39,74 +54,65 @@ This also speeds up the overall role.
+**WARNING: Some options only will work in AAP 2.6+**
 
-Options for the `gateway_role_user_assignments` variable:
+Options for the `aap_role_user_assignments` variable:
```
Have you actually updated the variable?

Tompage1994 commented on changelogs/fragments/aap_26_additions.yml:
```diff
@@ -0,0 +1,6 @@
+---
+minor_changes:
+  - Updated `gateway_role_user_assignments` role these changes will only work in AAP 2.6+
```
Have you actually changed the tasks? It doesn't seem to be the case, in which case it should still work in 2.5.

Ideally any change to this role will be backwards compatible to 2.5

djdanielsson replied:
so some options work with 2.5 but not all of them I believe

djdanielsson committed:
modified:   roles/gateway_role_definitions/README.md
	modified:   roles/gateway_role_definitions/meta/argument_specs.yml
	modified:   roles/gateway_role_team_assignments/README.md
	modified:   roles/gateway_role_team_assignments/meta/argument_specs.yml
	modified:   roles/gateway_role_user_assignments/README.md
	modified:   roles/gateway_role_user_assignments/meta/argument_specs.yml

sean-m-sullivan reviewed (approved):
LGTM, good changes
